bpf/bpf_network.c

```diff
@@ -44,8 +44,12 @@ static inline int handle_ipv6(struct __sk_buff *skb)
 	bool decrypted;
 
 	decrypted = ((skb->mark & MARK_MAGIC_HOST_MASK) == MARK_MAGIC_DECRYPT);
-	if (!revalidate_data(skb, &data, &data_end, &ip6))
-		return DROP_INVALID;
+	if (!revalidate_data(skb, &data, &data_end, &ip6)) {
+		int err = skb_pull_data(skb, ETH_HLEN + sizeof(struct iphdr));
+
+		if (err || !revalidate_data(skb, &data, &data_end, &ip4))
```

Comment:
This looks not correct, above should both be ipv6 hdr, no?

Reply:
Right

Reply:
my test cluster had ipv6 disabled so it didn't catch the compiler error above either. So will fix both and push update.